### PR TITLE
cleanup after the introduction of `bmcd`

### PR DESCRIPTION
**Pull request by svenrademakers**
this commit removes files and packages that are obsolete because of the new bmcd, intended to replace the old bmc.

### CODE REVIEW

File: tp2bmc/board/tp2bmc/overlay/bin/curl
Comment (ruslashev) [inline tp2bmc/board/tp2bmc/overlay/bin/curl:1]: Does a new `curl` binary from a package work? I tried to run `curl` on master but it was broken due to outdated SSL libraries. A new one should work or may need `BR2_PACKAGE_LIBCURL_OPENSSL`

File: .gitignore
Comment (ruslashev) [inline .gitignore:1]: `app/` is gone completely, and the rest is covered by `buildroot/.gitignore`

File: tp2bmc/board/tp2bmc/overlay/bin/curl
Comment (svenrademakers) [reply tp2bmc/board/tp2bmc/overlay/bin/curl:1]: `curl` is not a system-critical application of the firmware anymore. I'm unsure if we need it in our default configuration at all.

File: tp2bmc/configs/tp2bmc_defconfig [hunk 1]
```diff
@@ -88,7 +87,6 @@ BR2_PACKAGE_LIBAIO=y
 BR2_PACKAGE_LIBCURL_CURL=y
 BR2_PACKAGE_LIBCURL=y
 BR2_PACKAGE_LIBEVENT=y
-BR2_PACKAGE_LIBGPIOD_TOOLS=y
```
[the remark this answers is not in the document]
Comment (svenrademakers) [reply tp2bmc/configs/tp2bmc_defconfig:91]: `libgpiod` yes, `libgpiod_tools` not

**Review by ruslashev — APPROVED** [review]

**Pull request merged**